Service/Product/FacetBuilder.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service\Product;
+
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Magento\Customer\Api\GroupExcludedWebsiteRepositoryInterface;
+use Magento\Customer\Model\ResourceModel\Group\Collection as GroupCollection;
+use Magento\Directory\Model\Currency as CurrencyHelper;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+
+class FacetBuilder
```

Comment:
Wondering if it's worth adding a `FacetBuilderInterface` with a single `getAttributesForFaceting` method to implement, but I guess this will be only useful for products so that's not mandatory 🤷

Reply:
Maybe... I considered it but couldn't imagine other implementations of it at this time. I also didn't want to commit to the public API just yet - at least until the `dynamicWidgets` work is fully implemented.